File: 452-minimum-number-of-arrows-to-burst-balloons/minimum-number-of-arrows-to-burst-balloons.java
class Solution {
    public int findMinArrowShots(int[][] points) {
        Arrays.sort(points, (a, b) -> Integer.compare(a[1], b[1]));
        // Arrays.sort(points, (a, b) -> a[1] - b[1]);
        int pos = points[0][1], shots = 1;
        for (int i = 0; i < points.length; i++) {
            if (pos < points[i][0]) {
                shots++;
                pos = points[i][1];
            }
        }
        return shots;
    }
}

File: 452-minimum-number-of-arrows-to-burst-balloons/minimum-number-of-arrows-to-burst-balloons.py
class Solution:
    def findMinArrowShots(self, points: List[List[int]]) -> int:
        points.sort(key=lambda x: x[1])
        shot_pos = points[0][1]
        shots = 1
        for l, r in points:
            if shot_pos < l:
                shot_pos = r            # need another shot
                shots += 1
        return shots